ui/src/components/questions/post.js

```diff
@@ -0,0 +1,277 @@
+import moment from 'moment';
+import Certificate from '../utils/certificate';
+import * as User from '../../models/user';
+import Button from '../forms/button';
+import { UserContext } from '../../usercontext';
+import Trophy from '../utils/trophy';
+import { t } from 'ttag';
+import Dialog from '../utils/dialog';
+import InputField from '../forms/inputfield';
+
+class Post extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            votes: 0,
+            isDeleteDialogOpen: false,
+            isEditDialogOpen: false,
+            currentBody: '',
+            postingToServer: false
+        };
+    }
+
+    componentDidMount() {
+        this.setState({
+            votes: this.props.votes
+        });
+    }
+
+    getClasses() {
+        let classes = `question-post ${this.props.className}`;
+        if (this.state.votes < -3) {
+            classes = `${classes} faded`;
+        }
+        if (this.props.answered) {
+            classes = `${classes} accepted`;
+        }
+        return classes;
+    }
+
+    incrementVotes() {
+        this.setState({
+            votes: this.state.votes + 1
+        });
+    }
+
+    decrementVotes() {
+        this.setState({
+            votes: this.state.votes - 1
+        });
+    }
+
+    getTime() {
+        return moment(this.props.created).fromNow();
+    }
+
+    renderAnswered() {
+        return (
+            <Certificate active={this.props.answered} />
+        );
+    }
+
+    renderAwarded() {
+        const isTrophyActive = this.state.votes >= 20;
+        return <Trophy active={isTrophyActive} />;
+    }
+
+    onChangeAnswer(e) {
+        const value = e.target.value;
+        this.setState({
+            answer: value
+        });
+    }
+
+    closeEditDialog() {
+        this.setState({
+            isEditDialogOpen: false
+        });
+    }
+
+    openDeleteDialog() {
+        this.setState({
+            isDeleteDialogOpen: true
+        });
+    }
+
+    closeDeleteDialog() {
+        this.setState({
+            isDeleteDialogOpen: false
+        });
+    }
+
+    getDialogTitle() {
+        if (!this.props.questionId) {
+            return t`Delete answer`;
+        }
+
+        return t`Delete question`;
+    }
+
+    getDialogBody() {
+        let prefix = <b>{this.props.title}</b>;
+        if (!this.props.questionId) {
+            prefix = t`The answer`;
+        }
+
+        return <>{prefix} {t`will be deleted from Rocket fuel.`}</>;
+    }
+
+    editPost() {
+        if (!this.props.questionId) {
+            this.setState({
+                answer: this.props.body,
+                isEditDialogOpen: true
+            });
+            return;
+        }
+        this.props.history.push(`/create/thread/${this.props.questionId}`);
+    }
+
+    saveAnswer() {
+        if (!this.props.answerId) {
+            return;
+        }
+        this.setState({
+            postingToServer: true
```

Comment:
same for this, maybe go over all postingToServer

Reply:
Hook me up with a better name and I'll change it for you. Altho, I do think the name says pretty much what it does.